File: src/main/java/net/vidageek/games/auth/AuthController.java
package net.vidageek.games.auth;

import br.com.caelum.vraptor.Get;
import br.com.caelum.vraptor.Resource;
import br.com.caelum.vraptor.Result;

@Resource
final public class AuthController {

	private final Result result;
	private final Providers providers;

	public AuthController(Result result, Providers providers) {
		this.result = result;
		this.providers = providers;
		
	}
	
	@Get("/auth/provider/{name}")
	public void provider(String name) {
		result.redirectTo(providers.byName(name).applicationAuthoritionUrl());
	}
	
}

File: src/main/scala/net/vidageek/games/auth/AuthController.scala
package net.vidageek.games.auth

import br.com.caelum.vraptor.Get
import br.com.caelum.vraptor.Resource
import br.com.caelum.vraptor.Result
import net.vidageek.games.UserHost

@Resource
class AuthController(result: Result, providers: Providers, player: Player) {

  @Get(Array("/auth/provider/{name}"))
  def provider(name: String) {
    player.provider = providers.byName(name)
    result.redirectTo(player.provider.applicationAuthoritionUrl)
  }

  @Get(Array("/authorization", "/authorization/"))
  def authorization(oauth_token: String, oauth_verifier: String) {
    player.authorize(AuthorizationVerifier(oauth_verifier))
    result.redirectTo(classOf[UserHost]).home
  }

  def logout {
    player.logout
    result.redirectTo(classOf[UserHost]).home
  }
}